File: HelperJobby/client-app/src/DTOs/jobRelatetedDTOs/IncompleteJobSalaryDTO.ts
import {ShowPayByOptions} from "../../enums/modelDataEnums/ShowPayByOptions";
import {IncompleteJobDTO} from "./IncompleteJobDTO";
import {SalaryRates} from "../../enums/modelDataEnums/SalaryRates";

export interface IncompleteJobSalaryDTO {
    id: number;
    showPayByOption: ShowPayByOptions;
    minimalAmount: number;
    maximalAmount?: number;
    salaryRate: SalaryRates;
    meetsMinSalaryRequirement: boolean;
    incompleteJobId: number;
    incompleteJob: IncompleteJobDTO;
}

File: HelperJobby/DTOs/Job/IncompleteJobSalaryDTO.cs
using ApplicationDomain.Enums;

namespace HelperJobby.DTOs.Job;

public class IncompleteJobSalaryDTO
{
    public int Id { get; set; }
    public ShowPayByOptions ShowPayByOption { get; set; }
    public decimal MinimalAmount { get; set; }
    public decimal? MaximalAmount { get; set; }
    public SalaryRates SalaryRate { get; set; }
    public int IncompleteJobId { get; set; }
    public bool MeetsMinSalaryRequirement { get; set; }
    public IncompleteJobDTO IncompleteJob { get; set; }

}